garrysmod/lua/weapons/weapon_medkit.lua

```diff
@@ -18,135 +18,178 @@ SWEP.UseHands = true
 SWEP.Primary.ClipSize = 100
 SWEP.Primary.DefaultClip = 100
 SWEP.Primary.Automatic = false
-SWEP.Primary.Ammo = "none"
+SWEP.Primary.Ammo = ""
 
 SWEP.Secondary.ClipSize = -1
 SWEP.Secondary.DefaultClip = -1
 SWEP.Secondary.Automatic = false
-SWEP.Secondary.Ammo = "none"
+SWEP.Secondary.Ammo = ""
+
+SWEP.HoldType = "slam"
+
+SWEP.HealSound = Sound( "HealthKit.Touch" )
+SWEP.DenySound = Sound( "WallHealth.Deny" )
+
+SWEP.HealCooldown = 0.5
+SWEP.DenyCooldown = 1
 
 SWEP.HealAmount = 20 -- Maximum heal amount per use
-SWEP.MaxAmmo = 100 -- Maxumum ammo
+SWEP.HealRange = 64
 
-local HealSound = Sound( "HealthKit.Touch" )
-local DenySound = Sound( "WallHealth.Deny" )
+SWEP.AmmoRegenFrequency = 1 -- Number of seconds before each ammo regen
+SWEP.AmmoRegenAmount = 2 -- Amount of ammo refilled every AmmoRegenFrequency seconds
 
 function SWEP:Initialize()
 
-	self:SetHoldType( "slam" )
+	self:SetHoldType( self.HoldType )
+
+	if ( CLIENT ) then
+		self.AmmoDisplay = {
+			Draw = true,
+			PrimaryClip = 0
+		}
+	end
+
+end
 
-	if ( CLIENT ) then return end
+function SWEP:SetupDataTables()
 
-	timer.Create( "medkit_ammo" .. self:EntIndex(), 1, 0, function()
-		if ( self:Clip1() < self.MaxAmmo ) then self:SetClip1( math.min( self:Clip1() + 2, self.MaxAmmo ) ) end
-	end )
+	self:NetworkVar( "Float", 0, "NextAmmoRegen" )
+	self:NetworkVar( "Float", 1, "NextIdle" )
 
 end
 
 function SWEP:PrimaryAttack()
 
-	if ( CLIENT ) then return end
-
 	local owner = self:GetOwner()
+	local isplayer = SERVER and owner:IsPlayer()
 
-	if ( owner:IsPlayer() ) then
+	if ( isplayer ) then
 		owner:LagCompensation( true )
 	end
 
+	local startpos = owner:GetShootPos()
 	local tr = util.TraceLine( {
-		start = owner:GetShootPos(),
-		endpos = owner:GetShootPos() + owner:GetAimVector() * 64,
+		start = startpos,
+		endpos = startpos + owner:GetAimVector() * 64,

```

Comment:
I believe `self.HealRange` was intended to be used here

Reply:
You're right